test/framework/src/main/java/org/elasticsearch/search/geo/GeoPointShapeQueryTestCase.java

```diff
@@ -583,4 +585,101 @@ public void testQueryMultiPoint() throws Exception {
             assertEquals(0, searchHits.getTotalHits().value);
         }
     }
+
+    public void testQueryPointFromGeoJSON() throws Exception {
+        createMapping(defaultIndexName, defaultGeoFieldName);
+        ensureGreen();
+
+        String doc1 = """
+            {
+              "geo": {
+                "coordinates": [ -35, -25.0 ],
+                "type": "Point"
+              }
+            }""";
+        client().index(new IndexRequest(defaultIndexName).id("1").source(doc1, XContentType.JSON).setRefreshPolicy(IMMEDIATE)).actionGet();
+
+        Point point = new Point(-35, -25);
+        {
+            SearchResponse response = client().prepareSearch(defaultIndexName)
+                .setQuery(QueryBuilders.geoShapeQuery(defaultGeoFieldName, point))
+                .get();
+            SearchHits searchHits = response.getHits();
+            assertEquals(1, searchHits.getTotalHits().value);
+        }
+        {
+            SearchResponse response = client().prepareSearch(defaultIndexName)
+                .setQuery(QueryBuilders.geoShapeQuery(defaultGeoFieldName, point).relation(ShapeRelation.WITHIN))
+                .get();
+            SearchHits searchHits = response.getHits();
+            assertEquals(1, searchHits.getTotalHits().value);
+        }
+        {
+            SearchResponse response = client().prepareSearch(defaultIndexName)
+                .setQuery(QueryBuilders.geoShapeQuery(defaultGeoFieldName, point).relation(ShapeRelation.CONTAINS))
+                .get();
+            SearchHits searchHits = response.getHits();
+            assertEquals(1, searchHits.getTotalHits().value);
+        }
+        {
+            SearchResponse response = client().prepareSearch(defaultIndexName)
+                .setQuery(QueryBuilders.geoShapeQuery(defaultGeoFieldName, point).relation(ShapeRelation.DISJOINT))
+                .get();
+            SearchHits searchHits = response.getHits();
+            assertEquals(0, searchHits.getTotalHits().value);
+        }
+    }
+
+    public void testQueryPointFromMultiPoint() throws Exception {
```

Comment:
This test is a duplicate of the test `testQueryPointFromMultiPointFormats` but with the number of supported formats reduced to those supported by the GeoJSON parser (so removing point as double[] and lat,lon string). This is because the current implementation added point geojson to the point parser, instead of adding point parsing to the geojson parser. The question is whether we should do it the other way round. The consequence of adding point support to the GeoJSON parser is that `geo_shape` mappers will start understanding the alternative versions of point (double[] and lat,lon string). Is that a good or a bad consequence?

Reply:
I've generalized the test, keeping only test data separate (so geo_point tests all four formats, while geo_shape tests only WKT and GeoJSON).